scripts/environment-prevalidation.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+echo Checking for required commands
+needed_commands=(aws jq)
+for tool in "${needed_commands[@]}"; do
+        echo -n "Looking for ${tool}..."
+        if ! which "${tool}" > /dev/null; then
+                echo "ERROR: Required command ${tool} not found. Please install it or add it to the \$PATH"
+                exit 1
+        fi
+        echo "found"
+done
+echo "All required commands found"
+
+echo -n "Checking aws-cli authentication..."
+if ! aws iam list-users >/dev/null 2>&1 ; then
+        if ! [[ -f ~/.aws/credentials ]] ; then
+                echo "Could not authenticate with AWS. Please set up your ~/.aws.credentials file"
+                exit 1
+        fi
+
+        if [ ! -n "${AWS_PROFILE}" ]; then
+                echo "Could not authenticate with AWS. Please re-run this script with AWS_PROFILE defined"
+                exit 1
+        fi
+        echo "Could not authenticate with AWS but could not determine why"
+        exit 1
+fi
+echo OK
+
+echo "Validating AWS configuration prior to building/updating the environment"
+
+# Though we use the CDK to create a stack and manage the infra therein, there
+# are a couple of things which we need to have in place before doing that. One
+# example is the database (which can be, but need not be in the same VPC), and
+# another being values in the parameter store in order to authenticate with the
+# app database.
+needed_params=(
+    "YNR_AWS_S3_MEDIA_BUCKET" "YNR_AWS_S3_SOPN_REGION"
+    "YNR_AWS_S3_SOPN_BUCKET" "YNR_AWS_S3_MEDIA_REGION"
+    "DJANGO_SETTINGS_MODULE" "postgres_username" "postgres_host" "postgres_password"
+    "FQDN"
+)
```

Comment:
The rest of this script makes sense, but I think we should remove this bit.

At the moment, this is a list of the env vars that are set in the "Dev - YNR - DC" account, but that isn't a list of all the env vars needed to stand up a properly configured app and I don't fancy duplicating that full list in both the app settings and also this shell script.

The right way to handle this is that we should structure the settings file so it fails loudly to bootstrap the app if a required env var is not set. Then it is easy to diagnose "app is not working because required var is not set in param store" without duplicating this information in two places.